internal/commands/app/create_inputs_test.go

```diff
@@ -0,0 +1,272 @@
+package app
+
+import (
+	"errors"
+	"fmt"
+	"path"
+	"testing"
+
+	"github.com/10gen/realm-cli/internal/cloud/atlas"
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+	"github.com/10gen/realm-cli/internal/local"
+	"github.com/10gen/realm-cli/internal/utils/test/assert"
+	"github.com/10gen/realm-cli/internal/utils/test/mock"
+
+	"github.com/Netflix/go-expect"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+func TestAppCreateInputsResolve(t *testing.T) {
+	for _, tc := range []struct {
+		description string
+		inputs      createInputs
+		procedure   func(c *expect.Console)
+		test        func(t *testing.T, i createInputs)
+	}{
+		{
+			description: "with no flags set should prompt for just name and set realm.Location and deployment model to defaults",
+			procedure: func(c *expect.Console) {
+				c.ExpectString("App Name")
+				c.SendLine("test-app")
+				c.ExpectEOF()
+			},
+			test: func(t *testing.T, i createInputs) {
+				assert.Equal(t, "test-app", i.Name)
+				assert.Equal(t, flagDeploymentModelDefault, i.DeploymentModel)
+				assert.Equal(t, flagLocationDefault, i.Location)
+			},
+		},
+		{
+			description: "with a name flag set should prompt for nothing else and set realm.Location and deployment model to defaults",
```

Comment:
for tests that "don't prompt" things, lets write them in the style where we aren't even using the interactive/pseudo terminal.  they can simply just be unit tested with `nil` or simple `ui`s